reconcile/openshift_namespaces.py

```diff
@@ -55,57 +54,51 @@ def get_shard_namespaces() -> List[Dict[str, str]]:
     return list(namespaces.values())
 
 
-def check_ns_exists(spec: Mapping[str, str],
-                    oc_map: OC_Map) -> Tuple[Mapping[str, str], Any]:
+def manage_namespaces(spec: Mapping[str, str],
+                      oc_map: OC_Map, dry_run: bool) -> None:
     cluster = spec['cluster']
     namespace = spec['namespace']
-    try:
-        exists = oc_map.get(cluster).project_exists(namespace)
-        return spec, exists
-    except StatusCodeError as e:
-        msg = (
-            f'cluster: {cluster}, namespace: {namespace}, exception: {str(e)}'
-        )
-        logging.error(msg)
-
-    return spec, None
-
-
-def manage_projects(spec: Mapping[str, str],
-                    oc_map: OC_Map, dry_run: bool) -> None:
-    cluster = spec['cluster']
-    namespace = spec['namespace']
-    action = spec["action"]
+    desired_state = spec["desired_state"]
 
     oc = oc_map.get(cluster)
     if not oc:
         logging.log(level=oc.log_level, msg=oc.message)
         return None
 
-    if action == NS_ACTION_CREATE:
-        try:
-            logging.info(['create', cluster, namespace])
-            if not dry_run:
-                oc.new_project(namespace)
-        except StatusCodeError as e:
-            msg = (
-                f'cluster: {cluster}, namespace: {namespace}, '
-                f'exception: {str(e)}'
-            )
-            logging.error(msg)
+    exists = oc.project_exists(namespace)
 
-    elif action == NS_ACTION_DELETE:
+    if not exists and desired_state == NS_STATE_PRESENT:
+        logging.info(['create', cluster, namespace])
+        if not dry_run:
+            oc.new_project(namespace)
+    elif exists and desired_state == NS_STATE_ABSENT:
```

Comment:
You still have the typo in the log statement below. That's why I wanted the dispatch table. You DRY and thus make fewer mistakes.